extract_endpoint/tests/test_azure_utils.py

```diff
@@ -0,0 +1,70 @@
+import io
+import pytest
+from azure.storage import blob
+import azure_utils
+
+
+@pytest.fixture
+def sample_account() -> str:
+    return 'devstoreaccount1'
+
+
+@pytest.fixture
+def sample_key() -> str:
+    return 'Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw=='
+
+
+@pytest.fixture
+def sample_container() -> str:
+    return 'energuide-extracted-data'
+
+
+@pytest.fixture
+def sample_domain() -> str:
+    return 'http://127.0.0.1:10000/devstoreaccount1'
+
+
+@pytest.fixture
+def sample_storage_coordinates(sample_account: str,
+                               sample_key: str,
+                               sample_container: str,
+                               sample_domain: str) -> azure_utils.StorageCoordinates:
+    return azure_utils.StorageCoordinates(account=sample_account, key=sample_key,
+                                          container=sample_container, domain=sample_domain)
+
+
+@pytest.fixture
+def sample_block_blob_service(sample_storage_coordinates: azure_utils.StorageCoordinates) -> blob.BlockBlobService:
+    block_blob_service = blob.BlockBlobService(account_name=sample_storage_coordinates.account,
+                                               account_key=sample_storage_coordinates.key,
+                                               custom_domain=sample_storage_coordinates.domain)
+    container = sample_storage_coordinates.container
+    for filename in [blob.name for blob in block_blob_service.list_blobs(container)]:
+        block_blob_service.delete_blob(container, filename)
+    return block_blob_service
```

Comment:
This is better. To ensure consistency, ideally we should be sure that there are no side-effects from running a test. This deletion logic should be in a fixture teardown phase, instead of a fixture setup phase.

```
block_blob_service = blob.BlockBlobService(account_name=sample_storage_coordinates.account,
    account_key=sample_storage_coordinates.key, custom_domain=sample_storage_coordinates.domain)
yield block_blob_service

container = sample_storage_coordinates.container
for filename in [blob.name for blob in block_blob_service.list_blobs(container)]:
    block_blob_service.delete_blob(container, filename)
```

Reply:
👍 similar to what you did in the gitlaw code.